tests/drivers/gpio/gpio_basic_api/boards/apollo4p_evb.overlay

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ / {
+	resources {
```

Comment:
Is there a reason the test only covers the first set of GPIO pins and not the whole set of coverage?

Reply:
This test is called `GPIO 2-Pin Test` and does not support testing more than 2 pins. 
There is also a need to use the same gpio controller: 
https://github.com/zephyrproject-rtos/zephyr/blob/fb003d1aa80cac2db7ae87c83f862760f5c8e895/tests/drivers/gpio/gpio_basic_api/src/main.c#L23-L33
We can change the `gpio0_31` controller to another, but the results will be exactly the same.